pkg/controller/splunkforwarder/splunkforwarder_controller.go

```diff
@@ -138,7 +160,7 @@ func (r *ReconcileSplunkForwarder) Reconcile(request reconcile.Request) (reconci
 			}
 		} else if err != nil {
 			return reconcile.Result{}, err
-		} else if instance.CreationTimestamp.After(cmFound.CreationTimestamp.Time) {
+		} else if instance.CreationTimestamp.After(cmFound.CreationTimestamp.Time) || r.CheckGenerationVersionOlder(cmFound.GetAnnotations(), instance) {
 			updateDaemonSet = true // Recreate the DaemonSet whenever we update the ConfigMaps
 			err = r.client.Update(context.TODO(), configmap)
```

Comment:
You are updating the object here without returning, if the operator is interrupted here or an `err` is returned between here and the if stanza to delete the daemon set you'd never refresh the daemon set since `updateDaemonSet` would be lost. Can we move this update until later or have another way to notify the daemon set to refresh?

Reply:
we do, the genVersion will still be older for the DS. So on startup after a redeploy it will still fail that check and run the update on the DS